📖 Update install and development help (#1011)
executablebooks/mystmd: README.md

## Developing

> The `mystmd` libraries and command line tools are written in typescript, and require [node and npm](https://nodejs.org) for development locally. The `mystmd` package can be installed by users using pip or conda, which is a thin python wrapper around the mystmd bundle. If you have installed mystmd in using python, uninstall the python version or deactivate your environment to use the local development version.

Reviewer: ```suggestion
The `mystmd` libraries and command line tools are written in TypeScript, and require [NodeJS and NPM](https://nodejs.org) for local development. The `mystmd` package, which is a thin python wrapper around the `mystmd` bundle, can be installed by users using `pip` or `conda`. If you have already installed `mystmd` (e.g. via `pip` or `conda`), it is recommended that you uninstall it (or deactivate the relevant environment) before using the local development version.
```

> For local development, [clone the repository](https://github.com/executablebooks/mystmd) and install the dependencies using npm. You can then build the libraries (`npm run build`) and then optionally link to your globally installed `mystmd` in node using the `npm run link` command.

Reviewer: ```suggestion
For local development, [clone the repository](https://github.com/executablebooks/mystmd) and install the dependencies using npm. You can then build the libraries (`npm run build`) and then optionally link to your globally installed `mystmd` in node using the `npm link` command.
```

```shell
git clone git@github.com:executablebooks/mystmd.git
cd mystmd
npm install
npm run build

> npm run link

Reviewer 2: When I run `npm run link` I got the following error:

```shell
❯ npm -v
10.1.0
❯ node -v
v20.9.0
❯ npm run link
npm ERR! Missing script: "link"
npm ERR! 
npm ERR! Did you mean this?
npm ERR!     npm run lint # run the "lint" package script
npm ERR! 
npm ERR! To see a list of scripts, run:
npm ERR!   npm run

npm ERR! A complete log of this run can be found in: /Users/choldgraf/.npm/_logs/2024-03-20T16_31_12_437Z-debug-0.log
```

Could you advise?

Reviewer: ```suggestion
npm link
```

```

> These commands allow you to use the `myst` CLI in any directory as usual, and updates to the build are picked up when you rebuild. After making changes, you must rebuild the packages (run `npm run build` again after changing a file), which is done efficiently depending on how deep your change is in the dependency tree, after the build is complete, you can reuse the myst client.

Reviewer: ```suggestion
These commands allow you to use the `myst` CLI in any directory as usual, and updates to the build are picked up when you rebuild. After making changes, you must rebuild the packages (via `npm run build` in the top-level directory), which is done efficiently depending on how deep your change is in the dependency tree. After the build is complete, you can reuse the myst client.
```

Tests are also a helpful development tool, which don't require full rebuilding. You can run the entire test suite using `npm run test`. If you are working in a particular package, change your working directory and run the tests there, to run in watch mode use `npm run test:watch`.

> When contributing code, the continuous integration will run linting and formatting, you can run `npm run lint` and `npm run lint:format` locally to ensure they will pass. Developing in VSCode can be setup to show you changes in real time and fix formatting issues on save (extensions: [eslint](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint) and [prettier](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode)).

Reviewer: ```suggestion
When contributing code, the continuous integration will run linting and formatting. You can run `npm run lint` and `npm run lint:format` locally to ensure they will pass. If you are using the VSCode editor, it can be setup to show you changes in real time and fix formatting issues on save (extensions: [eslint](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint) and [prettier](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode)).
```
